src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +461,53 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Entering();
+
+            if (!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if (autoLogonConfigManager.RestartNeeded())
```

Comment:
I have added an additional argument (for run command). Will discuss with you on how we can pass it to the below layers (i.e., from agent to agentcapabilitiesprovider).

---
In reply to: [118270158](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118270158) [](ancestors = 118270158)